File: python/Jewels_and_Stones.py
# https://leetcode.com/problems/jewels-and-stones/description/

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewels_set = set(jewels)
        ret = 0
        for letter in stones:
            if letter in jewels_set:
                ret += 1
        return ret

File: java/leetcode/JewelsAndStones.java
package leetcode;

import java.util.HashSet;
import java.util.Set;

//문제링크: https://leetcode.com/problems/jewels-and-stones/description/

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Set<Character> jewelSet = new HashSet<>();
        for (char jewel : jewels.toCharArray())
            jewelSet.add(jewel);

        int cnt = 0;
        for (char stone : stones.toCharArray()) {
            if (jewelSet.contains(stone))
                cnt++;
        }

        return cnt;
    }
}
